Yank/repex.py

```diff
@@ -1,752 +1,1428 @@
-    (This is just an illustrative example; use ParallelTempering class for actual production parallel tempering simulations.)
+    Parallel tempering simulation of alanine dipeptide in implicit solvent (replica
+    exchange among temperatures). This is just an illustrative example; use ParallelTempering
+    class for actual production parallel tempering simulations.
+
+    Create the system.
 
-    >>> # Create test system.
-    >>> from openmmtools import testsystems
-    >>> testsystem = testsystems.AlanineDipeptideImplicit()
-    >>> [system, positions] = [testsystem.system, testsystem.positions]
-    >>> # Create thermodynamic states for parallel tempering with exponentially-spaced schedule.
-    >>> from simtk import unit
     >>> import math
-    >>> nreplicas = 3 # number of temperature replicas
-    >>> T_min = 298.0 * unit.kelvin # minimum temperature
-    >>> T_max = 600.0 * unit.kelvin # maximum temperature
-    >>> T_i = [ T_min + (T_max - T_min) * (math.exp(float(i) / float(nreplicas-1)) - 1.0) / (math.e - 1.0) for i in range(nreplicas) ]
-    >>> states = [ ThermodynamicState(system=system, temperature=T_i[i]) for i in range(nreplicas) ]
-    >>> import tempfile
-    >>> store_filename = tempfile.NamedTemporaryFile(delete=False).name + '.nc'
-    >>> # Create simulation.
-    >>> simulation = ReplicaExchange(store_filename)
-    >>> simulation.create(states, positions) # initialize the replica-exchange simulation
-    >>> simulation.minimize = False
-    >>> simulation.number_of_iterations = 2 # set the simulation to only run 2 iterations
-    >>> simulation.timestep = 2.0 * unit.femtoseconds # set the timestep for integration
-    >>> simulation.nsteps_per_iteration = 50 # run 50 timesteps per iteration
-    >>> simulation.run() # run the simulation
-    >>> del simulation # clean up
-
-    Extend the simulation
-
-    >>> simulation = ReplicaExchange(store_filename)
-    >>> simulation.resume()
-    >>> simulation.number_of_iterations = 4 # extend
-    >>> simulation.run()
+    >>> from simtk import unit
+    >>> from openmmtools import testsystems, states, mcmc
+    >>> testsystem = testsystems.AlanineDipeptideImplicit()
+
+    Create thermodynamic states for parallel tempering with exponentially-spaced schedule.
+
+    >>> n_replicas = 3  # Number of temperature replicas.
+    >>> T_min = 298.0 * unit.kelvin  # Minimum temperature.
+    >>> T_max = 600.0 * unit.kelvin  # Maximum temperature.
+    >>> temperatures = [T_min + (T_max - T_min) * (math.exp(float(i) / float(nreplicas-1)) - 1.0) / (math.e - 1.0)
+    ...                 for i in range(n_replicas)]
+    >>> thermodynamic_states = [states.ThermodynamicState(system=testsystem.system, temperature=T)
+    ...                         for T in temperatures]
+
+    Initialize simulation object with options. Run with a GHMC integrator.
+
+    >>> move = mcmc.GHMCMove(timestep=2.0*unit.femtoseconds, n_steps=50)
```

Comment:
It is! You can pass a list here and I made it so that they are tied to the thermodynamic states (not the replicas) since I thought we may want to slightly tweak the propagation with different degrees of decoupling.